src/Data/List/Relation/Binary/Sublist/Heterogeneous/Properties.agda

```diff
@@ -0,0 +1,359 @@
+  ∷-injectiveʳ P.refl = P.refl
+
+  ∷ʳ-injective : ∀ {y xs ys} {pxs qxs : Sublist R xs ys} →
+                 (Sublist R xs (y ∷ ys) ∋ y ∷ʳ pxs) ≡ (y ∷ʳ qxs) → pxs ≡ qxs
+  ∷ʳ-injective P.refl = P.refl
+
+
+module _ {a b r} {A : Set a} {B : Set b} {R : REL A B r} where
+
+  length-mono-Sublist-≤ : ∀ {as bs} → Sublist R as bs → length as ≤ length bs
+  length-mono-Sublist-≤ []        = z≤n
+  length-mono-Sublist-≤ (y ∷ʳ rs) = ℕₚ.≤-step (length-mono-Sublist-≤ rs)
+  length-mono-Sublist-≤ (r ∷ rs)  = s≤s (length-mono-Sublist-≤ rs)
+
+------------------------------------------------------------------------
+-- Conversion to and from Pointwise (proto-reflexivity)
+
+  fromPointwise : Pointwise R ⇒ Sublist R
+  fromPointwise []       = []
+  fromPointwise (p ∷ ps) = p ∷ fromPointwise ps
+
+  toPointwise : ∀ {as bs} → length as ≡ length bs →
+                Sublist R as bs → Pointwise R as bs
+  toPointwise {bs = []}     eq []         = []
+  toPointwise {bs = b ∷ bs} eq (r ∷ rs)   = r ∷ toPointwise (ℕₚ.suc-injective eq) rs
+  toPointwise {bs = b ∷ bs} eq (.b ∷ʳ rs) =
+    ⊥-elim $ ℕₚ.<-irrefl eq (s≤s (length-mono-Sublist-≤ rs))
+
+module _ {a b c r s t} {A : Set a} {B : Set b} {C : Set c}
+         {R : REL A B r} {S : REL B C s} {T : REL A C t} where
+
+  trans : Trans R S T → Trans (Sublist R) (Sublist S) (Sublist T)
+  trans rs⇒t []        []        = []
+  trans rs⇒t rs        (y ∷ʳ ss) = y ∷ʳ trans rs⇒t rs ss
+  trans rs⇒t (y ∷ʳ rs) (s ∷ ss)  = _ ∷ʳ trans rs⇒t rs ss
+  trans rs⇒t (r ∷ rs)  (s ∷ ss)  = rs⇒t r s ∷ trans rs⇒t rs ss
+
+module _ {a b r s e} {A : Set a} {B : Set b}
+         {R : REL A B r} {S : REL B A s} {E : REL A B e} where
+
+  antisym : Antisym R S E → Antisym (Sublist R) (Sublist S) (Pointwise E)
+  antisym rs⇒e []        []        = []
+  antisym rs⇒e (r ∷ rs)  (s ∷ ss)  = rs⇒e r s ∷ antisym rs⇒e rs ss
+  -- impossible cases
+  antisym rs⇒e (_∷ʳ_ {xs} {ys₁} y rs) (_∷ʳ_ {ys₂} {zs} z ss) =
+    ⊥-elim $ ℕₚ.<-irrefl P.refl $ begin
+    length (y ∷ ys₁) ≤⟨ length-mono-Sublist-≤ ss ⟩
+    length zs        ≤⟨ ℕₚ.n≤1+n (length zs) ⟩
+    length (z ∷ zs)  ≤⟨ length-mono-Sublist-≤ rs ⟩
+    length ys₁       ∎ where open ℕₚ.≤-Reasoning
+  antisym rs⇒e (_∷ʳ_ {xs} {ys₁} y rs) (_∷_ {y} {ys₂} {z} {zs} s ss)  =
+    ⊥-elim $ ℕₚ.<-irrefl P.refl $ begin
+    length (z ∷ zs) ≤⟨ length-mono-Sublist-≤ rs ⟩
+    length ys₁      ≤⟨ length-mono-Sublist-≤ ss ⟩
+    length zs       ∎ where open ℕₚ.≤-Reasoning
+  antisym rs⇒e (_∷_ {x} {xs} {y} {ys₁} r rs)  (_∷ʳ_ {ys₂} {zs} z ss) =
+    ⊥-elim $ ℕₚ.<-irrefl P.refl $ begin
+    length (y ∷ ys₁) ≤⟨ length-mono-Sublist-≤ ss ⟩
+    length xs        ≤⟨ length-mono-Sublist-≤ rs ⟩
+    length ys₁       ∎ where open ℕₚ.≤-Reasoning
+
+------------------------------------------------------------------------
+-- Various functions' outputs are sublists
+
+module _ {a b r} {A : Set a} {B : Set b} {R : REL A B r} where
+
+  tail-Sublist : ∀ {as bs} → Pointwise R as bs →
```

Comment:
These properties should all use the +/- notation rather than having `Sublist` in the name.

Reply:
I don't think they should: `Sublist (drop n xs) ys` is not the same as `Sublist xs ys -> Sublist (drop n xs) (drop n ys)`.

Reply:
Okay `tail` is special because it's in that first form. The rest are all in the second form though right?

Reply:
The other ones are in the shape `Sublist as bs -> Sublist (f as) bs`. The function is only
applied to the first one (it's a generalisation of `Sublist (f xs) xs` which is useful when
the `R` relation `Sublist` is built upon is not transitive.

I guess I forgot to generalise `tail-Sublist` too! :D

Reply:
Ah okay, makes sense now. This is probably one of the cases mentioned in #595 where the notation is too compact to be readable. It's not like they need to be compatible somewhere else and writing out the full types would definitely help!

Reply:
I'll inline the notations & add comments to the section header to explain
the distinction.

Reply:
What would your thoughts be on calling them `tail-sublist` or `tail-isSublist` rather than `tail-Sublist`? Nitpicky I know.

Reply:
I was reusing these names: [`take-⊆` et al.](https://agda.github.io/agda-stdlib/Data.List.Relation.Binary.Sublist.Propositional.Properties.html#5819).

Whatever we pick, we should also apply the same convention to things like
[`≤⇒pred≤ : ∀ {m n} → m ≤ n → pred m ≤ n`](https://agda.github.io/agda-stdlib/Data.Nat.Properties.html#9298)
(and document it).

Reply:
Hmm maybe you're right and `tail-Sublist` is fine...